rskj-core/src/test/java/co/rsk/remasc/RemascFederationProviderTest.java

```diff
@@ -30,81 +36,84 @@ class RemascFederationProviderTest {
         int federatorIndex = 0;
 
         ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
         when(activations.isActive(ConsensusRule.RSKIP415)).thenReturn(true);
 
-        FederationSupport federationSupport =  mock(FederationSupport.class);
-
         byte[] rskPublicKey = HexUtils.strHexOrStrNumberToByteArray(RSK_PUBLIC_KEY);
-
-        when(federationSupport.getFederatorPublicKeyOfType(federatorIndex, FederationMember.KeyType.RSK))
-                .thenReturn(rskPublicKey);
+        FederationSupport federationSupport =  mock(FederationSupport.class);
+        when(federationSupport.getActiveFederatorPublicKeyOfType(
+            federatorIndex,
+            FederationMember.KeyType.RSK
+        )).thenReturn(rskPublicKey);
 
         RemascFederationProvider provider = new RemascFederationProvider(activations, federationSupport);
 
         RskAddress actualRskAddress = provider.getFederatorAddress(federatorIndex);
 
-        Assertions.assertEquals(RSK_ADDRESS_FROM_RSK_PUBLIC_KEY, actualRskAddress.toHexString());
-        verify(federationSupport, times(1)).getFederatorPublicKeyOfType(federatorIndex, FederationMember.KeyType.RSK);
-        verify(federationSupport, never()).getFederatorBtcPublicKey(federatorIndex);
-
+        assertEquals(RSK_ADDRESS_FROM_RSK_PUBLIC_KEY, actualRskAddress.toHexString());
+        verify(federationSupport, times(1)).getActiveFederatorPublicKeyOfType(
+            federatorIndex,
+            FederationMember.KeyType.RSK
+        );
+        verify(federationSupport, never()).getActiveFederatorBtcPublicKey(federatorIndex);
     }
 
     private static RemascFederationProvider getRemascFederationProvider() {
-
         Genesis genesisBlock = new BlockGenerator().getGenesisBlock();
         BlockChainBuilder builder = new BlockChainBuilder().setTesting(true).setGenesis(genesisBlock);
         Blockchain blockchain = builder.build();
 
         ActivationConfig.ForBlock activations = ActivationConfigsForTest.all().forBlock(blockchain.getBestBlock().getNumber());
 
-        BridgeMainNetConstants bridgeMainNetConstants = BridgeMainNetConstants.getInstance();
+        final FederationSupport federationSupport = getFederationSupport(
+            builder.getRepository(),
+            blockchain,
+            activations
+        );
 
-        BridgeStorageProvider bridgeStorageProvider = new BridgeStorageProvider(
-                builder.getRepository(),
-                PrecompiledContracts.BRIDGE_ADDR,
-                bridgeMainNetConstants,
-                activations
+        return new RemascFederationProvider(
+            activations,
+            federationSupport
         );
+    }
 
-        FederationSupport federationSupport = new FederationSupport(bridgeMainNetConstants, bridgeStorageProvider, blockchain.getBestBlock(), activations);
+    private static FederationSupport getFederationSupport(Repository repository, Blockchain blockchain, ForBlock activations) {

```

Comment:
no big deal, but since we are setting the network as mainnet inside the method, it could be named `getMainnetFederationSupport`. Just an idea